osu.Game/Rulesets/Difficulty/Skills/Skill.cs

```diff
@@ -66,42 +84,43 @@ public void Process(DifficultyHitObject current)
         /// <summary>
         /// Saves the current peak strain level to the list of strain peaks, which will be used to calculate an overall difficulty.
         /// </summary>
-        public void SaveCurrentPeak()
+        private void saveCurrentPeak()
         {
-            if (Previous.Count > 0)
-                strainPeaks.Add(currentSectionPeak);
+            strainPeaks.Add(currentSectionPeak);
         }
 
         /// <summary>
         /// Sets the initial strain level for a new section.
         /// </summary>
         /// <param name="time">The beginning of the new section in milliseconds.</param>
-        public void StartNewSectionFrom(double time)
+        private void startNewSectionFrom(double time)
         {
             // The maximum strain of the new section is not zero by default, strain decays as usual regardless of section boundaries.
             // This means we need to capture the strain level at the beginning of the new section, and use that as the initial peak level.
-            if (Previous.Count > 0)
-                currentSectionPeak = GetPeakStrain(time);
+            currentSectionPeak = GetPeakStrain(time);
         }
 
         /// <summary>
         /// Retrieves the peak strain at a point in time.
         /// </summary>
         /// <param name="time">The time to retrieve the peak strain at.</param>
         /// <returns>The peak strain.</returns>
-        protected virtual double GetPeakStrain(double time) => CurrentStrain * strainDecay(time - Previous[0].BaseObject.StartTime);
+        protected virtual double GetPeakStrain(double time) => CurrentStrain * strainDecay(time - Previous[0].StartTime);
 
         /// <summary>
-        /// Returns the calculated difficulty value representing all processed <see cref="DifficultyHitObject"/>s.
+        /// Returns the calculated difficulty value representing all <see cref="DifficultyHitObject"/>s that have been processed up to this point.
         /// </summary>
         public double DifficultyValue()
         {
             double difficulty = 0;
             double weight = 1;
 
+            var finalStrainPeaks = strainPeaks.ToList(); // Copy list so we don't create side effects
+            finalStrainPeaks.Add(currentSectionPeak); // Add final peak since it never gets saved
```

Comment:
This can (and should) just use `CurrentStrainPeaks` now.

Reply:
Fixed in 2f8fd1ceea8d5787d04985b9a70e724ae76a45a4